### ISSUE TITLE
word header and footer

### ISSUE THREAD

**Issue by firulais** [evidenceprime/html-docx-js#8]
Hallo,

The html-docx works realy great. Could get realy good results till now.
What would be the way to add header and footer in the resulting word document?

- changing the .tpl file
- using the htmlDocx.asBlob(content, header, footer) function and adding html for header and footer options/parameters
- how to add page numbers to the footer

Thx for any hint,
regeads, Willi

**Comment by anowak** [comment]
Sorry for the delayed response. We haven't tried including headers, but according to [this documentation](http://officeopenxml.com/WPheaders.php), it would involve both changing the template and adding extra files to the resulting ZIP.

Let me know if you had any progress on this, otherwise I will try to give it a shot myself.

**Comment by wongjasont** [comment]
Hi,

Any news or progress about the header/footer feature? Thanks

**Comment by kolchy** [comment]
Header/footers would be awesome! Any progress on this?

**Comment by prasadKodeInCloud** [comment]
Hi @anowak,

Adding the header relationship requires core ZIP generating logic as you mentioned. [Here ](https://jsfiddle.net/kf3jvh44/)I added sample header relationship to document.tpl just to mention additional sections to be added.  Can you help us on adding headers and footers since its a basic requirement in most of the cases?  

Thanks!

**Comment by evalencia-lancera** [comment]
Hi!

Any news or progress on the resolution for this issue?

**Comment by hrsantiago** [comment]
It can be add to the .mht file.
http://sebsauvage.net/wiki/doku.php?id=word_document_generation

**Comment by saigithub1** [comment]
can i download svg images using html-docsx-js?

**Comment by saleswon** [comment]
@prasadKodeInCloud has anyone forked and added this feature, or does anyone have experience with adding this feature? I see that you added a sample, but I still have no clue where to start.

**Comment by Quixomatic** [comment]
@prasadKodeInCloud has anyone forked and added this feature, or does anyone have experience with adding this feature? I see that you added a sample, but I still have no clue where to start.

**Comment by anowak** [comment]
The problem is that Word (at least, the latest Word for Mac) doesn't seem to accept altchunks with HTML in either header or footer.

I haven't managed to make this [one work](http://sebsauvage.net/wiki/doku.php?id=word_document_generation), but I'm also skeptical regarding its compatibility with altchunks.

The bottom line is that the only way to have header/footer in the generated document would be to pass the content in WordProcessing ML, for example:

```xml
<w:hdr xmlns:w="http://schemas.openxmlformats.org/wordprocessingml/2006/main">
  <w:p>
    <w:pPr>
       <w:pStyle w:val="Header"/>
    </w:pPr>
    <w:r>
       <w:t>Header text</w:t>
    </w:r>
  </w:p>
</w:hdr>
```

To sum up -- it is possible, but header content cannot be specified using HTML. The easiest would be to use Word to create an empty document with a header, unzip the DOCX file and extract the definition of header/footer from it. Would such workflow suit your needs?

**Comment by Quixomatic** [comment]
It could suit our needs if altchunks is the limitation. Could it potentially accept plaintext for extremely simple headers/footers though?

**Comment by Linpter** [comment]
Hi ! Would anyone have an example of such way of doing ? It's the first time i'm playing with docx files and i'm not sure how to "extract the definition of header/footer from it". Thanks a lot

**Comment by nitinac** [comment]
Any update on this issue? I am successfully able to generate docx with header using html-docx-js.
I have mentioned header in the main html file like below

![image](https://user-images.githubusercontent.com/39834257/62697774-20cfdf00-b9f9-11e9-9aff-8cfc240122db.png)


and used mso-header: h1; style 

This is working and can view header on every page of generated document.
But my requirement is to restrict header on first page only. So I have tried to apply below styles

mso-title-page: yes; 
mso-first-header: h1;

But header is not visible. Does anyone know how to do this? Thanks in advance.

**Comment by Gopinath83** [comment]
Any one able to fix the Header/Footer or Is it possible to pass template document with header and footer which can only convert HTML to Word ?
@anowak - Could you please try and let us know?
We have been trying different ways to get the Header/Footer which includes image in Header.

Any help will be appreciated!
Thanks

**Comment by victorrwangrinan** [comment]
yes, it can be added to .mht file, but first page header is not working

**Comment by starsoul666** [comment]
> mso-header: h1

how to add footer?
